test/unit-test/mqtt_agent_utest.c

```diff
@@ -30,25 +30,80 @@
 #include "unity.h"
 
 /* Include paths for public enums, structures, and macros. */
-
+#include "mqtt_agent.h"
 #include "mock_core_mqtt.h"
+#include "mock_core_mqtt_state.h"
 
 /**
- * @brief MQTT client identifier.
+ * @brief Time at the beginning of each test. Note that this is not updated with
+ * a real clock. Instead, we simply increment this variable.
  */
-#define MQTT_CLIENT_IDENTIFIER                 "testclient"
+static uint32_t globalEntryTime = 0;
+
+/* ========================================================================== */
 
 /**
- * @brief A sample network context that we set to NULL.
+ * @brief A mocked timer query function that increments on every call.
  */
-#define MQTT_SAMPLE_NETWORK_CONTEXT            ( NULL )
+static uint32_t getTime( void )
+{
+    return globalEntryTime++;
+}
+/* ========================================================================== */
 
 /**
- * @brief Time at the beginning of each test. Note that this is not updated with
- * a real clock. Instead, we simply increment this variable.
+ * @brief A mocked send function to send command to agent.
  */
-static uint32_t globalEntryTime = 0;
+static bool( send )( AgentMessageContext_t * pMsgCtx,
+                     const void * pData,
+                     uint32_t blockTimeMs )
```

Comment:
This is formatted differently from how our functions usually are
```suggestion
static bool send( AgentMessageContext_t * pMsgCtx,
                  const void * pData,
                  uint32_t blockTimeMs )
```
Also, the name `send` is a big generic. `mockSend`?